src/icp_v2.cc

```diff
@@ -680,16 +662,26 @@ icpHandleUdp(int sock, void *)
         debugs(12, 4, "icpHandleUdp: FD " << sock << ": received " <<
                (unsigned long int)len << " bytes from " << from);
 
-#ifdef ICP_PACKET_DUMP
-
-        icpPktDump(buf);
-#endif
-
         if ((size_t) len < sizeof(icp_common_t)) {
             debugs(12, 4, "icpHandleUdp: Ignoring too-small UDP packet");
             break;
         }
 
+        auto pkt = reinterpret_cast<icp_common_t *>(buf);
+        struct in_addr a;
+        a.s_addr = pkt->shostid;
+        debugs(12, 2, "ICP Client remote=" << from << " FD " << sock);
+        debugs(12, 2, "ICP Client REQUEST:\n---------\n" <<
+               "opcode=(" << std::setw(3) << pkt->opcode << ")" << icp_opcode_str[pkt->opcode] <<
+               ", version=" << pkt->version <<
+               ", length=" << ntohs(pkt->length) <<
+               ", reqnum=" << std::setw(8) << ntohl(pkt->reqnum) <<
+               ", flags=(" << std::hex << std::setw(8) << ntohl(pkt->flags) << ")" <<
+               ", shostid=" << Ip::Address(a) <<
+               Debug::Extra <<
+               Raw(nullptr, (buf+sizeof(icp_common_t)), ntohs(pkt->length)).minLevel(9) <<
```

Comment:
Please name this field, especially since it will appear on a separate cache.log line! For example:

```suggestion
               Raw("payload", (buf+sizeof(icp_common_t)), ntohs(pkt->length)).minLevel(9) <<
```

Reply:
Done in 01eb43d